instrumentation/netty/netty-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/client/HttpClientResponseTracingHandler.java

```diff
@@ -25,8 +31,15 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
       return;
     }
 
-    if (msg instanceof HttpResponse) {
+    if (msg instanceof FullHttpResponse) {
       tracer().end(context, (HttpResponse) msg);
+    } else if (msg instanceof HttpResponse) {
+      // Headers before body has been sent, store them to use when finishing the span.
+      ctx.channel().attr(HTTP_RESPONSE).set((HttpResponse) msg);
+    } else if (msg instanceof LastHttpContent) {
+      // Not a FullHttpResponse so this is content that has been sent after headers. Finish the
+      // span using what we stored in attrs.
+      tracer().end(context, ctx.channel().attr(HTTP_RESPONSE).get());
```

Comment:
I find it confusing that you talk about "sending" here. This is _client_ response handler, it receives data, not sends.

Reply:
Yup thanks